### PR TITLE
Cross promotional modal dp

### PR DESCRIPTION
**Pull request by deeppandya**
Resolves : https://github.com/brave/brave-browser/issues/10571

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [ ] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [x] Android
  - [ ] iOS
  - [ ] Linux
  - [ ] macOS
  - [ ] Windows
- Verified that these changes pass automated tests (unit, browser, security tests) on
  - [ ] iOS
  - [ ] Linux
  - [ ] macOS
  - [ ] Windows
- [ ] Verified that all lint errors/warnings are resolved (`npm run lint`)
- [ ] Ran `git rebase master` (if needed).
- [ ] Ran `git rebase -i` to squash commits (if needed).
- [ ] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.
- [ ] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone
- [ ] Public documentation has been updated as necessary. For instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-compatibility-issues-with-tracking-protection
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide

## Test Plan:
Follow the test plan on the issue and follow https://www.figma.com/file/atoL4LN2kd8ILnPGwohUoV/Mobile-onboarding-v2?node-id=697%3A5848 for the design spec

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on.
- [ ] All relevant documentation has been updated.

### CODE REVIEW

File: browser/ui/android/strings/android_brave_strings.grd [hunk 1]
```diff
@@ -1166,6 +1166,15 @@ until they verify, or until 90 days have passed.
       <message name="IDS_NICE" desc="Shields tootltip title">
         Nice!
       </message>
+      <message name="IDS_BRAVE_WORKS_EVERYWHERE" desc="Cross promotional modal title text">
+        Brave works everywhere
+      </message>
+      <message name="IDS_DOWNLOAD_BRAVE" desc="Cross promotional modal text">
+        Download Brave at brave.com on your laptop ot tablet for a faster and more private web
```
Comment (samartnik) [inline browser/ui/android/strings/android_brave_strings.grd:1173]: `laptop or tablet`?

File: browser/ui/android/strings/android_brave_strings.grd [hunk 2]
```diff
@@ -1166,6 +1166,15 @@ until they verify, or until 90 days have passed.
...
+      </message>
+      <message name="IDS_BRAVE_URL" desc="Brave url">
+        brave.com
```
Comment (samartnik) [inline browser/ui/android/strings/android_brave_strings.grd:1176]: Not sure if we need it as string here as it doesn't require translation

File: android/java/org/chromium/chrome/browser/CrossPromotionalModalDialogFragment.java [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+
+        Spanned textToSpan = BraveRewardsHelper.spannedFromHtmlString(mPromoText);
+        SpannableString ss = new SpannableString(textToSpan.toString());
+
+        ForegroundColorSpan foregroundSpan =
+            new ForegroundColorSpan(getResources().getColor(R.color.shield_back_button_tint));
+        ss.setSpan(foregroundSpan, indexOfSpan,
+                   (indexOfSpan + getResources().getString(R.string.brave_url).length()), Spanned.SPAN_EXCLUSIVE_EXCLUSIVE);
+        mCrossPromoText.setText(ss);
+
```
Comment (samartnik) [inline android/java/org/chromium/chrome/browser/CrossPromotionalModalDialogFragment.java:80]: nit: excessive line

File: android/java/org/chromium/chrome/browser/CrossPromotionalModalDialogFragment.java [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+            if (isLandscape) {
+                params.width = (int) (0.5 * mDeviceWidth);
+            } else {
+                params.width = (int) (0.9 * mDeviceWidth);
+            }
+        }
+        params.height = LinearLayout.LayoutParams.WRAP_CONTENT;
+        getDialog().getWindow().setAttributes(params);
+    }
+}
```
Comment (samartnik) [inline android/java/org/chromium/chrome/browser/CrossPromotionalModalDialogFragment.java:109]: nit: i chromium they add empty line in java files, I think we should it too for consistency

File: browser/ui/android/strings/android_brave_strings.grd (hunk 1 above)
Comment (deeppandya) [reply browser/ui/android/strings/android_brave_strings.grd:1173]: oops sorry. it's a typo. i will update it

File: browser/ui/android/strings/android_brave_strings.grd (hunk 2 above)
Comment (deeppandya) [reply browser/ui/android/strings/android_brave_strings.grd:1176]: i agree. i will remove it from strings and add it in the class as constant

Comment (SergeyZhukovsky) [reply browser/ui/android/strings/android_brave_strings.grd:1176]: Where do we use it?

**Pull request merged**